pkgs/games/osu-lazer/bin.nix

```
@@ -1,37 +1,84 @@
-{ appimageTools, lib, fetchurl }:
+{ lib
+, stdenv
+, fetchurl
+, fetchzip
+, appimageTools
+}:
 
-appimageTools.wrapType2 rec {
+let
   pname = "osu-lazer-bin";
-  version = "2023.301.0";
+  version = "2023.305.0";
+  name = "${pname}-${version}";
 
-  src = fetchurl {
-    url = "https://github.com/ppy/osu/releases/download/${version}/osu.AppImage";
-    sha256 = "sha256-0c74bGOY9f2K52xE7CZy/i3OfyCC+a6XGI30c6hI7jM=";
+  file = {
+    aarch64-darwin = "osu.app.Apple.Silicon.zip";
+    x86_64-darwin = "osu.app.Intel.zip";
+    x86_64-linux = "osu.AppImage";
+  }.${stdenv.system} or (throw "${pname}-${version}: ${stdenv.system} is unsupported.");
+
+  fileHash = {
+    aarch64-darwin = "sha256-vfDEEU+InD5uIc1QKceM2SN1GlWV93DySn3ik1MSYf4=";
+    x86_64-darwin = "sha256-dK+qqwBzuPXHqQk8U3TpWQBWEz+8S9GHShVwTTq2BBQ=";
+    x86_64-linux = "sha256-W3XJ7HtJM5iFI8OOTTu8IBHMerZSCETHMemkoTislK8=";
+  }.${stdenv.system};
```

More of a personal preference, but I would have both information in the same attrset. Something like:

```suggestion
  file = {
    aarch64-darwin = {
      name = "osu.app.Apple.Silicon.zip";
      hash = "sha256-vfDEEU+InD5uIc1QKceM2SN1GlWV93DySn3ik1MSYf4="
    };
    x86_64-darwin = {
      name = "osu.app.Intel.zip";
      hash = "sha256-dK+qqwBzuPXHqQk8U3TpWQBWEz+8S9GHShVwTTq2BBQ=";
    };
    x86_64-linux = {
      name = "osu.AppImage";
      hash = "sha256-W3XJ7HtJM5iFI8OOTTu8IBHMerZSCETHMemkoTislK8=";
    };
  }.${stdenv.system} or (throw "${pname}-${version}: ${stdenv.system} is unsupported.");
```